### ISSUE TITLE
Updating activity code to work from 3.1.1 to 4.0.1

### ISSUE THREAD

**Issue by wizzkidd** [brianwernick/ExoMedia#453]
- [ ✓] I have verified there are no duplicate active or recent bugs, questions, or requests

###### Include the following:
 - ExoMedia version: `4.0.1`
 - Device OS version: `7.0`
 - Devide Manufacturer: `Google`
 - Device Name: `Pixel XL`
 
###### Reproduction Steps
I've changed my gradle dependencies from `compile 'com.devbrackets.android:exomedia:3.1.1'` to `compile 'com.devbrackets.android:exomedia:4.0.1'` and noticed the only major alterations required in my code was to use `VideoView `instead of `EMVideoView`.  Also the layout xml needed to use `VideoView `instead of `EMVideoView`.  At compile time, I noticed several errors that are pointing at a DemoPlayer.java file.  The DemoPlayer.java file, if memory serves me right, was ripped from [this project](https://github.com/3sidedcube/Android-ExoPlayer/blob/master/demo/src/main/java/com/google/android/exoplayer/demo/full/player/DemoPlayer.java).  I appreciate @brianwernick that you are not the author of that class, however I'm asking for some assistance to either remove the necessity of using it, or alternatively bumping up the code within it to work with the 4.0.1 release.

At a glance, the DemoPlayer.java references to imports such as:
```
import com.google.android.exoplayer.CodecCounters;
import com.google.android.exoplayer.DummyTrackRenderer;
import com.google.android.exoplayer.ExoPlaybackException;
import com.google.android.exoplayer.ExoPlayer;
import com.google.android.exoplayer.MediaCodecAudioTrackRenderer;
import com.google.android.exoplayer.MediaCodecTrackRenderer;
```

Could you assist in any way?  My guess is that these need to be altered to something like `com.google.android.exoplayer2.xxx.yyy.zzz` ?

I have a method in my code that utilises the DemoPlayer.java class as shown below:
```
private static XmlTvProgram parseProgram(XmlPullParser parser)
            throws IOException, XmlPullParserException, ParseException {
        String channelId = null;
        Long startTimeUtcMillis = null;
        Long endTimeUtcMillis = null;
        String videoSrc = null;
        int videoType = DemoPlayer.SOURCE_TYPE_HTTP_PROGRESSIVE;  //** DemoPlayer IS USED HERE **//
        for (int i = 0; i < parser.getAttributeCount(); ++i) {
            String attr = parser.getAttributeName(i);
            String value = parser.getAttributeValue(i);
            if (ATTR_CHANNEL.equalsIgnoreCase(attr)) {
                channelId = value;
            } else if (ATTR_START.equalsIgnoreCase(attr)) {
                startTimeUtcMillis = DATE_FORMAT.parse(value).getTime();
            } else if (ATTR_STOP.equalsIgnoreCase(attr)) {
                endTimeUtcMillis = DATE_FORMAT.parse(value).getTime();
            } else if (ATTR_VIDEO_SRC.equalsIgnoreCase(attr)) {
                videoSrc = value;
            } else if (ATTR_VIDEO_TYPE.equalsIgnoreCase(attr)) {
                if (VALUE_VIDEO_TYPE_HTTP_PROGRESSIVE.equals(value)) {
                    videoType = DemoPlayer.SOURCE_TYPE_HTTP_PROGRESSIVE;
                } else if (VALUE_VIDEO_TYPE_HLS.equals(value)) {
                    videoType = DemoPlayer.SOURCE_TYPE_HLS;
                } else if (VALUE_VIDEO_TYPE_MPEG_DASH.equals(value)) {
                    videoType = DemoPlayer.SOURCE_TYPE_MPEG_DASH;
                }
            }
        }
...
...
...
}
```
It's not usually like me to ask for spoon feeding, however the problem with copy and pasting chunks of code is that when there are significant changes to a library, I have found it difficult to make the necessary changes to the ripped class.  So I really appreciate any input and support.  Thanks.

###### Expected Result
Successful compilation using the 4.0.1 version of ExoMedia

###### Actual Result
Numerous errors at compile time suggesting the import code is incorrect.

**Comment by se-bastiaan** [comment]
Why do you need that DemoPlayer.java file exactly? Isn't it redundant if your use ExoMedia?

**Comment by wizzkidd** [comment]
@se-bastiaan You are most likely correct in that it may be redundant, however since I ripped the class from an existing project as described above, it is being used in just one of my methods and to be honest, I dont know how I would work around it.

**Comment by brianwernick** [comment]
What you are seeing is expected as ExoMedia `3.y.z` used ExoPlayer `1.5.z` while ExoMedia `4.0.z` uses ExoPlayer `2.y.z` which has a changed path (among many other changes).
What is it you need from that particular `DemoPlayer` class that isn't provided directly from either ExoPlayer or ExoMedia?

**Comment by wizzkidd** [comment]
@brianwernick Thanks for taking the time to answer my question.

It would seem that my code is setting an integer for `videoType `as seen here:
`int videoType = DemoPlayer.SOURCE_TYPE_HTTP_PROGRESSIVE;`

I then have a condition that respectively sets `videoType `as seen here:
```
if (VALUE_VIDEO_TYPE_HTTP_PROGRESSIVE.equals(value)) {
                    videoType = DemoPlayer.SOURCE_TYPE_HTTP_PROGRESSIVE;
                } else if (VALUE_VIDEO_TYPE_HLS.equals(value)) {
                    videoType = DemoPlayer.SOURCE_TYPE_HLS;
                } else if (VALUE_VIDEO_TYPE_MPEG_DASH.equals(value)) {
                    videoType = DemoPlayer.SOURCE_TYPE_MPEG_DASH;
                }
```
A snippet of the method is shown in my original post for reference.  But from what I can make out, the could be detecting the type of video, and setting the integer variable accordingly.  I have just attempted to bypass this by locating the line of code that uses the videoType value further on, and I have simply hardcoded it to be '1'.  My app will compile successfully now, but ideally i'd like to see if I can avoid the hardcoded value.
